src/test/java/com/mentoring/RegistrationTest.java

```diff
@@ -31,4 +38,88 @@ public void successfulRegistration() {
         //TODO: Add notification assertion
         softAssertion.assertAll();
     }
+
+    // FIXME: modify this solution to a globally working one
+    // prerequisite:
+    // a.) query an existing e-mail address from db: "SELECT email FROM USER LIMIT 1"
+    // b.) successfulRegistration: save the given e-mail address to a global variable -> test case depends on another one -> WRONG
+    // c.) we assume that an exact e-mail address is in the database and we use that one
+    @Test(groups = "smoke")
+    public void alreadyRegisteredEmail() {
+        MainPage mainPage = new MainPage(driver)
+                .fillRegistrationForm(User.generateRandomUsername(), VALID_PASSWORD, EXISTING_EMAIL);
+        softAssertion.assertThat(mainPage.invalidEmailAddress())
+                .as("X icon (validation error) should be visible at e-mail address.")
+                .isTrue();
+        softAssertion.assertThat(!mainPage.isSubmitButtonEnabled())
+                .as("Submit button should be disabled.")
+                .isTrue();
+        softAssertion.assertAll();
+    }
+
+    @Test(groups = "smoke")
+    public void emptyRegistration() {
+        MainPage mainPage = new MainPage(driver)
+                .fillRegistrationForm("", "", "");
+        softAssertion.assertThat(mainPage.invalidUserName())
+                .as("X icon (validation error) should be visible at username.")
+                .isTrue();
+        softAssertion.assertThat(mainPage.invalidPassword())
+                .as("X icon (validation error) should be visible at password.")
+                .isTrue();
+        softAssertion.assertThat(mainPage.invalidEmailAddress())
+                .as("X icon (validation error) should be visible at e-mail address.")
+                .isTrue();
+        softAssertion.assertThat(!mainPage.isSubmitButtonEnabled())
+                .as("Submit button should be disabled.")
+                .isTrue();
+        softAssertion.assertAll();
+    }
+
+    @Test(groups = "smoke")
+    public void wrongEmailAddress() {
+        MainPage mainPage = new MainPage(driver)
+                .fillRegistrationForm(User.generateRandomUsername(), VALID_PASSWORD, INVALID_EMAIL);
+        softAssertion.assertThat(mainPage.invalidEmailAddress())
+                .as("X icon (validation error) should be visible at e-mail address.")
+                .isTrue();
+        softAssertion.assertThat(!mainPage.isSubmitButtonEnabled())
+                .as("Submit button should be disabled.")
+                .isTrue();
+        softAssertion.assertAll();
+    }
+
+    @Test(groups = "smoke")
+    public void tooShortFields() {
+        MainPage mainPage = new MainPage(driver)
+                .fillRegistrationForm(TOO_SHORT_USERNAME, TOO_SHORT_PASSWORD, User.generateRandomEmail());
```

Comment:
I suggest here to go with the same approach as I commented in emptyRegistration() test.